modules/hardware/polarfire/configuration.nix

```diff
@@ -0,0 +1,107 @@
+{ config, pkgs, modulesPath, lib, ...}:
+let
+   #User configuration
+   #To enable docker, make sure all required kernel configs are enabled
+   enable-docker = false;
+   blacklisted-kernel-modules = [];
+   system-packages = [];
+
+   #Private
+   docker-system-packages =  if enable-docker then [ pkgs.nftables ] else [];
+   blacklist-iptables = if enable-docker then ["ip_tables"] else [];
+   blacklisted-modules = blacklisted-kernel-modules ++ blacklist-iptables;
+   all-system-packages = system-packages ++ docker-system-packages;
+in
+{
+  imports = [
+    (modulesPath + "/profiles/minimal.nix")
+    (modulesPath + "/installer/sd-card/sd-image.nix")
+    ../../development/authentication.nix
+    ../../development/ssh.nix
+  ];
+
+  nixpkgs = {
+       localSystem.config = "x86_64-unknown-linux-gnu";
+       crossSystem.config = "riscv64-unknown-linux-gnu";
+  };
+
+
+  /* system version */
+  /* ============== */
+
+  #system.stateVersion = "23.05";
+  system.stateVersion = "22.11";
+
+
+  /* Boot configuration */
+  /* ================== */
+  
+  boot.blacklistedKernelModules = blacklisted-modules;
+  boot.kernelParams = [ "root=/dev/mmcblk0p2" "rootdelay=5"  ];
+  boot.consoleLogLevel = 4;
+  boot.loader = {
+    grub.enable = false;
+    generic-extlinux-compatible.enable = true;
+  };
+
+  /* General */
+  /* ======= */
+
+  disabledModules = [ "profiles/all-hardware.nix" ];
+  #required for alfred
+  environment.noXlibs = false;
+  environment.systemPackages = all-system-packages;
+
+
+  /* Networking */
+  /* ========== */
+
+  networking.hostName = "ghaf-host";
```

Comment:
This should be used from common definitions for reference configuration - see https://github.com/tiiuae/ghaf/blob/44a8660de954404c5ce1b47d31f89136b941ec97/modules/host/default.nix#L19
Hardware specifics should not override / conflict / duplicate with the common `networking.hostName` - this will keep a ghaf system on different hardware more familiar for developers/users.

Reply:
Created a new varian of networking.nix for such platforms which doesn't use vm.